docs/source/design/ui_design.rst

```diff
@@ -0,0 +1,151 @@
+
+Below you can find how to run an SCF calculation for a hydrogen molecule using
+QCEngine.
+
+.. code-block:: python
+
+    import qcengine as qcng
+    import qcelemental as qcel
+    mol = qcel.models.Molecule.from_data('H 0. 0. 0. \n H 0. 0. 1.')
+    out = qcng.compute({"molecule": mol, "driver": "energy", "model": {"method": "SCF", "basis": "sto-3g"}})
+    energy = out.return_result
+
+Here, mol is the molecule object (type
+``qcelemental.models.molecule.Molecule``), which is created using the
+``qcel.models.Molecule.from_data()`` function from QCElemental package. The SCF
+energy is computed using the ``qcng.compute()`` function, from QCEngine
+package. Note that, the input for the `qcng.compute` function is a Python
+dictionary with a schema defined by QCElemental.
+
+
+NWChemEx
+========
+
+In NWChemEx, running an SCF calculation for a hydrogen molecule is made easy
+through the use of a Python function with named arguments, as shown below.
+
+.. code-block:: python
+
+    import nwchemex as nwx 
+    energy = nwx.calculate_scf_energy(molecule = 'H 0. 0. 0. \n H 0. 0. 1.', basis = 'sto-3g')
+
+While the example shows two options for the ``nwx.calculate_scf_energy()``
+function, the whole function signature is given below: 
+
+.. code-block:: python
+
+    def calculate_scf_energy(molecule: Union[str, chemist.Molecule], basis:
+    Union[str, simde.type.ao_space], spin: int = 1, max_iterations: int = 50,
+    thresh: float = 1e-10, module_manager: PlugingPlay.ModuleManager = None, 
+    options: Union[Dict, dataclass] = None, **kwargs) -> float:
+
+In this function, the only required argument is ``molecule``, which can either
+be a Python ``string`` or a ``chemist.Molecule`` object. Optional ``basis``
+argument can either be a a Python ``string`` or a ``simde.type.ao_space``, with
+the minimal basis set "sto-3g" chosen as the default value. Additional SCF
+related options are also explicitly defined in the function with reasonable
+default values. ``module_manager`` option is reserved for power users, who would
+like to provide an existing module_manager object. While the number of options are
+limited for SCF, for correlated methods more options may need to be defined on
+top of SCF options. One solution to avoid a long list of arguments is to provide
+additional options through ``**kwargs`` option at the end. However, this would
+require users to know the correct keywords. To help our users to figure out the
+available keywords, we also provide the ``options`` argument, which is a Python
+dataclass holding all the possible options with reasonable default values. Our
+users can grab this data structure and figure out the available options and
+their default values and modify them as they wish. With ``options`` set
+explicitly, users can also call a more general ``nwx.calculate_energy``
+function, where the method is specified as an option.
+
+
+.. code-block:: python
```

Comment:
Okay so this is the main API? If so, maybe start with this API and introduce the SCF as a simplification/specialization.